pyomo/contrib/incidence_analysis/tests/test_interface.py

```diff
@@ -1791,6 +1791,57 @@ def test_linear_only(self):
         self.assertIs(matching[m.eq2], m.x[2])
         self.assertIs(matching[m.eq3], m.x[3])
 
+    def test_add_edge(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var([1, 2, 3, 4])
+        m.eq1 = pyo.Constraint(expr=m.x[1] ** 2 + m.x[2] ** 2 + m.x[3] ** 2 == 1)
+        m.eq2 = pyo.Constraint(expr=m.x[2] + pyo.sqrt(m.x[1]) + pyo.exp(m.x[3]) == 1)
+        m.eq3 = pyo.Constraint(expr=m.x[3] + m.x[2] + m.x[4] == 1)
+        m.eq4 = pyo.Constraint(expr=m.x[1] + m.x[2] ** 2 == 5)
+
+        # nodes: component
+        # 0    : eq1
+        # 1    : eq2
+        # 2    : eq3
+        # 3    : eq4
+        # 4    : x[1]
+        # 5    : x[2]
+        # 6    : x[3]
+        # 7    : x[4]
+
+        igraph = IncidenceGraphInterface(m, linear_only=False)
+        n_edges_original = igraph.n_edges
+
+        # Test if there already exists an edge between two nodes, nothing is added
+        igraph.add_edge_to_graph(m.eq3, m.x[4])
+        n_edges_new = igraph.n_edges
+        self.assertEqual(n_edges_original, n_edges_new)
+
+        igraph.add_edge_to_graph(m.x[1], m.eq3)
+        n_edges_new = igraph.n_edges
+        self.assertEqual(set(igraph._incidence_graph[2]), {6, 5, 7, 4})
+        self.assertEqual(n_edges_original + 1, n_edges_new)
+
+        igraph.add_edge_to_graph(m.eq4, m.x[4])
+        n_edges_new = igraph.n_edges
+        self.assertEqual(set(igraph._incidence_graph[3]), {4, 5, 7})
```

Comment:
As above, use `igraph.get_adjacent_to` instead of accessing `igraph._incidence_graph`.